docs/config-file.md

```diff
@@ -12,30 +12,84 @@ export default async function () {
 }
 ```
 
-This function must return an object with the following shape:
+This function must return an object with protocol specific configuration which would look like this: 
 
+
+```js
+export default async function () {
+  return {
+    websocket: {},
+    mqtt: {},
+    cluster: {}
+  }
+}
+
+```
+
+Here is an example of the `glee.config.js` file for refence: 
 ```js
 export default async function () {
   return {
     websocket: {
-      httpServer: customServer, // A custom HTTP server of your own.
-      adapter: 'native', // Default. Can also be 'socket.io' or a reference to a custom adapter.
-      port: process.env.PORT,
+      server: {
+        httpServer: customServer, // A custom HTTP server of your own.
+        adapter: "native", // Default. Can also be 'socket.io' or a reference to a custom adapter.
+        port: process.env.PORT,
+      },
+      client: {
+        authentication: {
+          http: () => {
+            const token = process.env.TOKEN
+            return token
+          }
```

Comment:
```suggestion
          http: () => process.env.TOKEN
```

Reply:
BTW, [WebSocket authentication is always HTTP](https://www.rfc-editor.org/rfc/rfc6455#section-4) so it makes more sense to use `token` instead of `http`.

Reply:
I have removed the client example as well, and will be tackling it in #319